blocks/api/registration.js

```diff
@@ -73,6 +80,24 @@ export function registerBlockType( name, settings ) {
 		);
 		return;
 	}
+	if ( ! ( 'category' in settings ) ) {
+		console.error(
+			'The block "' + name + '" must have a category.'
+		);
+		return;
+	}
+	if ( 'category' in settings && typeof settings.category !== 'string' ) {
+		console.error(
+			'The category slug for block "' + name + '" must be a string.'
+		);
+		return;
+	}
+	if ( 'category' in settings && ! ( categories.find( x => x.slug === settings.category ) ) ) {
```

Comment:
We should avoid `.find` since these are not polyfilled, Could we replace this with lodash's `find`?